driver.go

```diff
@@ -49,8 +66,16 @@ type Pin interface {
 	Name() string
 	Number() int
 }
+
 type Driver interface {
 	io.Closer
 	Metadata() Metadata
```

Comment:
Keeping Metadata in Driver since it is used in reef-pi for pin maps and other errors.